examples/naive/aarch64/intt_dilithium_1234_5678_manual_ld4.s

```diff
@@ -0,0 +1,512 @@
+        ldr_vo data9, in, (9*(512/8))
+        ldr_vo data10, in, (10*(512/8))
+        ldr_vo data11, in, (11*(512/8))
+        ldr_vo data12, in, (12*(512/8))
+        ldr_vo data13, in, (13*(512/8))
+        ldr_vo data14, in, (14*(512/8))
+        ldr_vo data15, in, (15*(512/8))
+
+        // layer4
+        gs_butterfly data0, data1, root3, 2, 3
+        gs_butterfly data2, data3, root4, 0, 1
+        gs_butterfly data4, data5, root4, 2, 3
+        gs_butterfly data6, data7, root5, 0, 1
+        gs_butterfly data8, data9, root5, 2, 3
+        gs_butterfly data10, data11, root6, 0, 1
+        gs_butterfly data12, data13, root6, 2, 3
+        gs_butterfly data14, data15, root7, 0, 1
+
+        // layer3
+        gs_butterfly data0, data2, root1, 2, 3
+        gs_butterfly data1, data3, root1, 2, 3
+        gs_butterfly data4, data6, root2, 0, 1
+        gs_butterfly data5, data7, root2, 0, 1
+        gs_butterfly data8, data10, root2, 2, 3
+        gs_butterfly data9, data11, root2, 2, 3
+        gs_butterfly data12, data14, root3, 0, 1
+        gs_butterfly data13, data15, root3, 0, 1
+
+        // layer2
+        gs_butterfly data0, data4, root0, 2, 3
+        gs_butterfly data1, data5, root0, 2, 3
+        gs_butterfly data2, data6, root0, 2, 3
+        gs_butterfly data3, data7, root0, 2, 3
+        gs_butterfly data8, data12, root1, 0, 1
+        gs_butterfly data9, data13, root1, 0, 1
+        gs_butterfly data10, data14, root1, 0, 1
+        gs_butterfly data11, data15, root1, 0, 1
+
+        // layer 1
+        gs_butterfly data0, data8, root0, 0, 1
+        gs_butterfly data1, data9, root0, 0, 1
+        gs_butterfly data2, data10, root0, 0, 1
+        gs_butterfly data3, data11, root0, 0, 1
+        gs_butterfly data4, data12, root0, 0, 1
+        gs_butterfly data5, data13, root0, 0, 1
+        gs_butterfly data6, data14, root0, 0, 1
+        gs_butterfly data7, data15, root0, 0, 1
+
+        canonical_reduce data8,  modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data9,  modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data10, modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data11, modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data12, modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data13, modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data14, modulus_half, neg_modulus_half, t2, t3
+        canonical_reduce data15, modulus_half, neg_modulus_half, t2, t3
+
+        str_vo data8, in,  (8*(512/8))
+        str_vo data9, in,  (9*(512/8))
+        str_vo data10, in, (10*(512/8))
+        str_vo data11, in, (11*(512/8))
+        str_vo data12, in, (12*(512/8))
+        str_vo data13, in, (13*(512/8))
+        str_vo data14, in, (14*(512/8))
+        str_vo data15, in, (15*(512/8))
+
+        mul_ninv data8, data9, data10, data11, data12, data13, data14, data15, data0, data1, data2, data3, data4, data5, data6, data7
```

Comment:
Oof, excuse my ignorance, but can you remind me why we need this (algorithmically)?

Reply:
Ah, ok. It's the usual scaling by `1/N`, but we only need to do it for half of the output because for the other half, it has already been incorporated into the twiddles.

Could you add a comment to that effect?